File: btvn_b8/app/src/main/java/com/example/btvn_b8/MainActivity.java
package com.example.btvn_b8;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.widget.Button;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;

public class MainActivity extends AppCompatActivity {

    private CounterViewModel viewModel;
    private SharedPreferences prefs;
    private TextView tvCount;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        tvCount = findViewById(R.id.tvCount);

        Button btnIncrement = findViewById(R.id.btnIncrement);
        Button btnDecrement = findViewById(R.id.btnDecrement);
        Button btnReset = findViewById(R.id.btnReset);

        prefs = getSharedPreferences(
                "counter_prefs",
                MODE_PRIVATE
        );

        viewModel = new ViewModelProvider(this)
                .get(CounterViewModel.class);

        viewModel.loadCount(prefs);

        viewModel.getCount().observe(this, count -> {
            tvCount.setText(String.valueOf(count));
        });

        btnIncrement.setOnClickListener(v ->
                viewModel.increment(prefs)
        );

        btnDecrement.setOnClickListener(v ->
                viewModel.decrement(prefs)
        );

        btnReset.setOnClickListener(v ->
                viewModel.reset(prefs)
        );

    }
}

File: btvn_b5/app/src/main/java/com/example/btvn_b5/MainActivity.kt
package com.example.btvn_b5


import android.os.Bundle
import android.widget.Button
import android.widget.ImageView
import androidx.appcompat.app.AppCompatActivity


class MainActivity : AppCompatActivity() {


    private lateinit var imageView: ImageView
    private lateinit var btnNext: Button
    private lateinit var btnBack: Button


    private val images = arrayOf(
        R.drawable.anh1,
        R.drawable.anh2,
        R.drawable.anh3
    )


    private var currentIndex = 0


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_main)


        imageView = findViewById(R.id.imageView)
        btnNext = findViewById(R.id.btnNext)
        btnBack = findViewById(R.id.btnBack)


        showImage()


        btnNext.setOnClickListener {

            if (currentIndex < images.size - 1) {

                currentIndex++

                showImage()

            }

        }


        btnBack.setOnClickListener {

            if (currentIndex > 0) {

                currentIndex--

                showImage()

            }

        }


        updateButton()

    }


    private fun showImage(){

        imageView.setImageResource(images[currentIndex])

        updateButton()

    }


    private fun updateButton(){



        btnBack.visibility =
            if(currentIndex == 0)
                Button.INVISIBLE
            else
                Button.VISIBLE




        btnNext.visibility =
            if(currentIndex == images.size - 1)
                Button.INVISIBLE
            else
                Button.VISIBLE

    }


}